src/libraries/System.Private.Uri/src/System/IriHelper.cs

```diff
@@ -231,7 +206,7 @@ internal static unsafe string EscapeUnescapeIri(char* pInput, int start, int end
                     if ((char.IsHighSurrogate(ch)) && (next + 1 < end))
                     {
                         ch2 = pInput[next + 1];
-                        escape = !CheckIriUnicodeRange(ch, ch2, ref surrogatePair, component == UriComponents.Query);
+                        escape = !CheckIriUnicodeRange(ch, ch2, out surrogatePair, component == UriComponents.Query);
```

Comment:
Changing this from `ref` to `out` forces the caller to do the indirect write.  Why is that a perf improvement?

Reply:
This specifically is not about performance.

When I look at a method that takes a ref I assume that it is also interested in the current value. CheckIriUnicodeRange is not, it never reads it and always assigns it, so out is more appropriate here imo from a readability standpoint.

Reply:
> This specifically is not about performance.

Ok, but such a change can negatively impact performance.  It may not in this particular case, but I expect perf is exactly why it was written this way in the first place.  It's a relatively common pattern.

Reply:
Do you think it's worth changing back over readability in this case?

Reply:
Only if you can detect a measurable impact on perf, in particular in the case where it's not a surrogate pair.

Reply:
This method will only be called if the first character has been checked to be a high surrogate.

Even for an input like `{ '\ud83f', 'a' } * 1000`, I can't see a measurable difference.